akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala

```diff
@@ -524,6 +524,9 @@ private[akka] trait ClusterRouterActor { this: RouterActor =>
     satisfiesRoles(m.roles) &&
     (settings.allowLocalRoutees || m.address != cluster.selfAddress)
 
+  private def memberShuttingDown(event: MemberEvent): Boolean =
```

Comment:
Why isn't `isAvailable` enough? It checks status Up or WeaklyUp. I thought the status would be `PreparingForShutdown` or `ReadyForShutdown` for those new events?

Reply:
Indeed, I may have trusted the bug report a bit too blindly, it should work out of the box.